### ISSUE TITLE
Enhance `ferium upgrade` by splitting into `update`, `upgrade`, `pin`, `unpin` (similarly to package managers)

### ISSUE THREAD

**Issue by T3sT3ro** [gorilla-devs/ferium#256]
## The problem
I made this issue to consolidate the propositions in #163, #162 and #49 by proposing a common solution.

TL;DR of the issues in question:
- do a partial upgrade (like `ferium upgrade REI`)
- show a list of upgradable mods without downloading
- show a selection of mods to download
- use regex/glob pattern for mod names

## Solutions
I propose to make the interface of ferium similar to `apt` linux package manager (optional arguments in square brackets):
- `ferium update [mods...]` - fetches information about new possible versions WITHOUT downloading new versions. Prints the list of possible upgrades followed by:
  > "To upgrade all mods use `ferium upgrade`"
  
  If `[mods...]` is empty, fetch update of all mods
- `ferium upgrade [mods...]` - downloads latest mods. If `[mods...]` is empty, upgrade all mods. Otherwise match mods with regex/glob pattern. This can be made interactive by using currently present interactive prompt:
  - without mod names it prints `[Y/n]?` next to each mod to confirm downloading of versions
  - with `-y` flag it automatically installs all mods without prompting
  - with `[mods...]` it never asks for confirmation with a prompt

In this case there is no need for `--force` or `--dry-run` flags, because the interface described above is standard and well known

Additionally 2 new commands could be added to assist with the upgrade process

- `ferium pin <mods...>` - add a `pinned` flag in the profile to disallow upgrading of the mod (updates are possible, only upgrades are not). Useful when trying to lock mod to some stable version
- `ferium unpin <mods...>` - remove above `pinned` flag.

In this case whenever mod list is displayed (or `ferium update` run) the flag is displayed as nect to the mod name in a form of icon or text `[pinned]` next to version, to indicate that the mod cannot be upgraded unless stated otherwise.

## Why?
Sometimes major versions can change for a mod (like create from 0.4 to 0.5) removing and changing a lot of content, thus potentially breaking the game for the unsuspecting. It would be nice to have more control on what is and isn't upgraded. I added pinning because I assume it's a very simple change with adding one flag to the profile, but it gives a lot of control.

**Comment by theRookieCoder** [comment]
I prefer to have issues split. It's really a matter of personal preference, I prefer it because I can implement features slowly instead of having it seems like I'm leaving enhancements partially developed. (It's also psychologically better to split things into subtasks)

Your suggested interface is standard in package managers, but I think `--force` and `--dry-run` are simpler and more intuitive.

Your 'pinning' suggestion is related to #82

For package managers that maintain a local database/store of the versions and packages, separate `update` and `upgrade` commands make a lot of sense. However in ferium's case it just doesn't make sense, and resolving versions is so fast it doesn't matter anyways. In my opinion, the `--dry-run` flag is enough for this functionality.
